file-formats/enhs/type/zif_seef_aff_enhs_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+INTERFACE zif_seef_aff_enhs_v1
+  PUBLIC.
+
+  "! <p class="shorttext">BAdI Enhancement Spot Tool Type</p>
+  "! Tool type of the BAdI Enhancement Spot
+  "! $values {@link zif_seef_aff_enhs_v1.data:co_tool}
+  TYPES ty_tool TYPE c LENGTH 10.
+
+  CONSTANTS:
+    "! <p class="shorttext">Tool Type</p>
+    "! Enhancement Spot tool type
+    BEGIN OF co_tool,
+      "! <p class="shorttext">BAdI Definition</p>
+      "! BAdI definition
+      badi_definition TYPE ty_tool VALUE 'BADI_DEF',
+    END OF co_tool.
+
+  "! <p class="shorttext">Instantiation of the BAdI Definition</p>
+  "! Instantiation of the BAdI definition
+  "! $values {@link zif_seef_aff_enhs_v1.data:co_instantiation}
+  TYPES ty_instantiation TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Instantiation of the BAdI Definition</p>
+    "! Instantiation of the BAdI definition
+    BEGIN OF co_instantiation,
+      "! <p class="shorttext">Creating New Instances</p>
+      "! Creating new instances
+      creating_new_instances     TYPE ty_instantiation VALUE 'N',
+      "! <p class="shorttext">Reuse Instances</p>
+      "! Reuse instances
+      reuse_instances            TYPE ty_instantiation VALUE 'S',
+      "! <p class="shorttext">Context-Specific Instances</p>
+      "! Context-specific instances
+      context_specific_instances TYPE ty_instantiation VALUE 'Y',
+    END OF co_instantiation.
+
+  "! <p class="shorttext">Filter Type</p>
+  "! Type of a badi filter
+  "! $values {@link zif_seef_aff_enhs_v1.data:co_filter_type}
+  TYPES ty_filter_type TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Filter Type</p>
+    "! Type of the filter
+    BEGIN OF co_filter_type,
+      "! <p class="shorttext">Integer</p>
+      "! Integer
+      integer        TYPE ty_filter_type VALUE 'I',
+      "! <p class="shorttext">Character-Like</p>
+      "! Character-like
+      character_like TYPE ty_filter_type VALUE 'C',
+      "! <p class="shorttext">String</p>
+      "! String
+      string         TYPE ty_filter_type VALUE 'S',
+      "! <p class="shorttext">Numeric</p>
+      "! Numeric
+      numeric        TYPE ty_filter_type VALUE 'N',
+      "! <p class="shorttext">Packed</p>
+      "! Packed
+      packed         TYPE ty_filter_type VALUE 'P',
+    END OF co_filter_type.
+
+  "! <p class="shorttext">Type of the Value Check Object</p>
+  "! Type of the value check object
+  "! $values {@link zif_seef_aff_enhs_v1.data:co_value_check_object_type}
+  TYPES ty_value_check_object_type TYPE string.
+
+  CONSTANTS:
+    "! Type of the value check object
+    "! $values {@link zif_seef_aff_enhs_v1.data:co_value_check_object_type}
```

Comment:
We don't need the annotation `$values` here, do we?

Reply:
That's why a developer should avoid copy/paste...